### PR TITLE
Improving analytics filter

### PR DESCRIPTION
**Pull request by dinusha92**
## Purpose
Improving analytics filter by adding a file location to write files.
Omit filter when throttle out or request is auth filter failed.

## Goals
File location added to use it as a shared location.

### CODE REVIEW

File: components/micro-gateway-cli/src/main/resources/distribution/bin/gateway [hunk 1]
```diff
@@ -141,4 +141,4 @@ elif [ "$CMD" = "restart" ]; then
 fi
 
 # run the balx created for ${label} APIs
-sh $BALLERINA_HOME/bin/ballerina run $GWHOME/exec/${label}.balx -e b7a.http.accesslog.path=$GWHOME/logs/access_logs --config $GWHOME/conf/micro-gw.conf "$@"
+sh $BALLERINA_HOME/bin/ballerina run $GWHOME/exec/${label}.balx -e api.usage.data.path=$GWHOME/api-usage-data -e b7a.http.accesslog.path=$GWHOME/logs/access_logs --config $GWHOME/conf/micro-gw.conf
```
Comment (chamilaadhi) [inline components/micro-gateway-cli/src/main/resources/distribution/bin/gateway]: You have removed "$@" part . Please add it back :D

File: components/micro-gateway-core/src/main/ballerina/gateway/executiontime_util.bal [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import ballerina/log;
```
Comment (pubudu538) [inline components/micro-gateway-core/src/main/ballerina/gateway/executiontime_util.bal]: License header is missing. Please check all the files in the PR.

File: components/micro-gateway-core/src/main/ballerina/gateway/executiontime_util.bal [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        + executionTimeDTO.securityLatency + OBJ + executionTimeDTO.throttlingLatency + OBJ + executionTimeDTO.
+        requestMediationLatency + OBJ +
+        executionTimeDTO.responseMediationLatency + OBJ + executionTimeDTO.backEndLatency + OBJ + executionTimeDTO.
+        otherLatency + OBJ + executionTimeDTO.eventTime;
+
+    return output;
+}
+
+function getMetaDataForExecutionTimeDTO(ExecutionTimeDTO dto) returns string {
+    return "{\\\"keyType\\\":\"" + dto.keyType + "\",\\\"correlationID\\\":\"" + dto.correleationID + "\"}";
```
Comment (pubudu538) [inline components/micro-gateway-core/src/main/ballerina/gateway/executiontime_util.bal]: Can't we create a Json payload and convert to string in Ballerina?

File: components/micro-gateway-cli/src/main/resources/distribution/bin/gateway (hunk 1 above)
Comment (chamilaadhi) [inline components/micro-gateway-cli/src/main/resources/distribution/bin/gateway]: You have removed "$@" part . Please add it back :D

File: components/micro-gateway-core/src/main/ballerina/gateway/executiontime_util.bal (hunk 2 above)
Comment (dinusha92) [reply components/micro-gateway-core/src/main/ballerina/gateway/executiontime_util.bal]: Added

File: components/micro-gateway-core/src/main/ballerina/gateway/executiontime_util.bal (hunk 3 above)
Comment (dinusha92) [reply components/micro-gateway-core/src/main/ballerina/gateway/executiontime_util.bal]: Fixed in  https://github.com/wso2/product-microgateway/pull/148

**Pull request merged**